python/tests/test_kmeans.py

```diff
@@ -63,14 +63,52 @@ def assert_centers_equal(
         assert a_center == pytest.approx(b_center, tolerance)
 
 
-def test_params() -> None:
+@pytest.mark.parametrize("default_params", [True, False])
+def test_params(default_params: bool) -> None:
     from cuml import KMeans as CumlKMeans
 
+    spark_params = {
+        "k": 2,
+        "initMode": "k-means||",
+        "tol": 0.0001,
+        "maxIter": 20,
+    }
+
     cuml_params = get_default_cuml_parameters(
-        [CumlKMeans], ["handle", "output_type", "convert_dtype"]
+        cuml_classes=[CumlKMeans], excludes=["handle", "output_type", "convert_dtype"]
     )
-    spark_params = KMeans()._get_cuml_params_default()
-    assert cuml_params == spark_params
+
+    # Ensure internal cuml defaults match actual cuml defaults
+    assert KMeans()._get_cuml_params_default() == cuml_params
+
+    # Our algorithm overrides the following cuml parameters with their spark defaults:
+    spark_default_overrides = {
+        "n_clusters": 2,
+        "max_iter": 20,
+        "init": "k-means||",
+    }
+
+    for param, value in spark_default_overrides.items():
+        if param in cuml_params:
+            cuml_params[param] = value
+
+    if default_params:
+        kmeans = KMeans()
+        seed = kmeans.getSeed()  # get the random seed that Spark generates
+        cuml_params["random_state"] = seed
+    else:
+        kmeans = KMeans(
+            n_clusters=10,
```

Comment:
Seems for the overlapping algos with mllib we should use spark param names to initialize here (and below.)